src/components/AppLayout/Header/components/Notifications.tsx

```diff
@@ -1,47 +1,310 @@
-import { ReactElement, useMemo } from 'react'
-import { useSelector } from 'react-redux'
-import { IconButton, Badge } from '@material-ui/core'
+import { ReactElement, useMemo, useRef, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { IconButton, Badge, ClickAwayListener, Grow, Paper, Popper } from '@material-ui/core'
 import NotificationsNoneIcon from '@material-ui/icons/NotificationsNone'
 import styled from 'styled-components'
+import { OptionsObject } from 'notistack'
 
 import { ReturnValue as Props } from 'src/logic/hooks/useStateHandler'
-import { selectNotifications } from 'src/logic/notifications/store/notifications'
-import { primary400 } from 'src/theme/variables'
+import {
+  deleteAllNotifications,
+  NotificationsState,
+  readNotification,
+  selectNotifications,
+} from 'src/logic/notifications/store/notifications'
+import { black300, black500, border, primary200, primary400, sm } from 'src/theme/variables'
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
+import ExpandLessIcon from '@material-ui/icons/ExpandLess'
+import { formatTime } from 'src/utils/date'
+import AlertIcon from 'src/assets/icons/alert.svg'
+import CheckIcon from 'src/assets/icons/check.svg'
+import ErrorIcon from 'src/assets/icons/error.svg'
+import InfoIcon from 'src/assets/icons/info.svg'
+import { StyledScrollableBar } from 'src/routes/safe/components/Transactions/TxList/styled'
+
+const NOTIFICATION_LIMIT = 4
+
+export const getSortedNotifications = (notifications: NotificationsState): NotificationsState => {
+  const unreadActionNotifications: NotificationsState = []
+  const unreadNotifications: NotificationsState = []
+  const readNotifications: NotificationsState = []
+
+  const chronologicalNotifications = notifications.sort((a, b) => b.timestamp - a.timestamp)
+
+  for (const notification of chronologicalNotifications) {
+    if (notification.read) {
+      readNotifications.push(notification)
+      continue
+    }
+
+    if (notification.action) {
+      unreadActionNotifications.push(notification)
+    } else {
+      unreadNotifications.push(notification)
+    }
+  }
+
+  return unreadActionNotifications.concat(unreadNotifications, readNotifications)
```

Comment:
Oh, you're just contacting them all again?

Please use one single sort instead.
`_. orderBy(notifications, ['timestamp', 'read', 'action' ])`.